src/SWP/Bundle/ContentBundle/Controller/RoutesController.php

```diff
@@ -0,0 +1,252 @@
+     *     statusCodes={
+     *         201="Returned on success."
+     *     },
+     *     input="SWP\Bundle\ContentBundle\Form\Type\RouteType"
+     * )
+     * @Route("/api/{version}/content/routes/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_content_create_routes")
+     * @Method("POST")
+     */
+    public function createAction(Request $request)
+    {
+        $manager = $this->get('doctrine_phpcr')->getManager();
+
+        $form = $this->createForm(new RouteType(), [], ['method' => $request->getMethod()]);
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            $formData = $form->getData();
+            if (!isset($formData['parent']) || is_null($formData['parent'])) {
+                $formData['parent'] = '/';
+            }
+
+            $route = $this->handleRouteUpdate(new Document\Route(), $formData);
+            $manager->persist($route);
+            $manager->flush();
+
+            $this->get('event_dispatcher')
+                ->dispatch(HttpCacheEvent::EVENT_NAME, new HttpCacheEvent($route));
+
+            return $this->handleView(View::create($route, 201));
+        }
+
+        return $this->handleView(View::create($form, 200));
+    }
+
+    /**
+     * Updates routes for current tenant.
+     *
+     * Parameter `type` cane have one of two values: `content` or `collection`.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Updates routes for current tenant",
+     *     statusCodes={
+     *         200="Returned on success."
+     *     },
+     *     input="SWP\Bundle\ContentBundle\Form\Type\RouteType"
+     * )
+     * @Route("/api/{version}/content/routes/{id}", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_content_update_routes", requirements={"id"=".+"})
+     * @Method("PATCH")
+     */
+    public function updateAction(Request $request, $id)
+    {
+        $manager = $this->get('doctrine_phpcr')->getManager();
+        $routeBasepath = $this->get('swp_multi_tenancy.path_builder')
+            ->build($this->getParameter('swp_multi_tenancy.persistence.phpcr.base_paths')[0]);
+        $route = $manager->find('SWP\Bundle\ContentBundle\Document\Route', $routeBasepath.$id);
+        if (!$route) {
+            throw new NotFoundHttpException('Route was not found.');
+        }
+
+        $form = $this->createForm(new RouteType(), [], ['method' => $request->getMethod()]);
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            $this->get('event_dispatcher')
+                ->dispatch(HttpCacheEvent::EVENT_NAME, new HttpCacheEvent($route));
+
+            $route = $this->handleRouteUpdate($route, $form->getData());
+            $manager->flush();
+
+            return $this->handleView(View::create($route, 200));
+        }
+
+        return $this->handleView(View::create($form, 200));
+    }
+
+    private function handleRouteUpdate($route, $routeData)
```

Comment:
we will move it to service when we will need to use it in any other place. For now it's safe here as a private method.